src/archive.rs

```diff
@@ -242,117 +270,158 @@ impl Archive {
+                .map(|block_id| block_dir.compressed_size(block_id).unwrap_or_default())
+                .sum();
+
+            monitor_.measure_unreferenced_blocks_finished();
+            total_bytes
+        };
+
         stats.unreferenced_block_bytes = total_bytes;
 
         if !options.dry_run {
             delete_guard.check()?;
 
-            let progress = nutmeg::View::new(
-                LinearModel::new("Delete bands", delete_band_ids.len()),
-                ui::nutmeg_options(),
-            );
-            for band_id in delete_band_ids {
-                Band::delete(self, band_id)?;
-                stats.deleted_band_count += 1;
-                progress.update(|model| model.increment(1));
+            {
+                let mut delete_band_count = 0;
+                monitor_.delete_bands(0, delete_band_ids.len());
+                for band_id in delete_band_ids {
+                    delete_band_count += 1;
+                    monitor_.delete_bands(delete_band_count, delete_band_ids.len());
+
+                    Band::delete(self, band_id)?;
+                    stats.deleted_band_count += 1;
+                }
+                monitor_.delete_bands_finished();
             }
 
-            let progress = nutmeg::View::new(
-                LinearModel::new("Delete blocks", unref_count),
-                ui::nutmeg_options(),
-            );
-            let error_count = unref
-                .par_iter()
-                .inspect(|_| progress.update(|model| model.increment(1)))
-                .filter(|block_hash| block_dir.delete_block(block_hash).is_err())
-                .count();
-            stats.deletion_errors += error_count;
-            stats.deleted_block_count += unref_count - error_count;
+            {
+                monitor_.delete_blocks(0, unref.len());
+
+                let delete_block_count = AtomicUsize::new(0);
+                let error_count = unref
+                    .par_iter()
+                    .inspect(|_| {
+                        monitor_.delete_blocks(
+                            delete_block_count.fetch_add(1, Ordering::Relaxed) + 1,
+                            unref.len(),
+                        );
+                    })
+                    .filter(|block_hash| block_dir.delete_block(block_hash).is_err())
+                    .count();
+
+                stats.deletion_errors += error_count;
+                stats.deleted_block_count += unref_count - error_count;
+                monitor_.delete_blocks_finished();
+            }
         }
 
         stats.elapsed = start.elapsed();
         Ok(stats)
     }
 
-    pub fn validate(&self, options: &ValidateOptions) -> Result<ValidateStats> {
+    pub fn validate(
+        &self,
+        options: &ValidateOptions,
+        monitor: Option<&dyn ValidateMonitor>,
+    ) -> Result<ValidateStats> {
+        let default_monitor = DefaultMonitor {};
+        let monitor = monitor.unwrap_or(&default_monitor);
```

Comment:
`NullMonitor` is a way better name. Using `unwrap_or_default` isn't possible, since we're dealing with a reference.  
That's why the default monitor is being created even if there has been an monitor provided.  
  
But refactoring DefaultMonitor into a static variable would be possible now (previously monitors were `&mut self`).